src/guide/3-behaviour-definition/4-agent-communication.rst

```diff
@@ -5,13 +5,195 @@ Communication between agents in FLAMEGPU2 is handled through messages. Messages
 
 *TODO: Diagram demonstrating messaging structure*
 
-
 Defining Messages
 -----------------
+Defining a message type in FLAMEGPU2 requires selection of a communication strategy and specification of the data the message will contain. FLAME GPU comes 
+with several built-in communication strategies, described below. It can also be extended to support bespoke messaging types. For guidance on this see the file 
+``include/flamegpu/runtime/messaging.h``. For each message type, the communication strategy defines how the messages will be accessed.
+
+============== =========================== ==================================================
+Type           Symbol                       Description
+============== =========================== ==================================================
+Brute Force    ``MsgBruteForce``           Access all messages
+Spatial 2D     ``MsgSpatial2D``            Access all messages within a radius in 2D
+Spatial 3D     ``MsgSpatial3D``            Access all messages within a radius in 3D
+Array 1D       ``MsgArray1D``              Directly access messages via a 1 dimensional array
+Array 2D       ``MsgArray2D``              Directly access messages via a 2 dimensional array
+Array 3D       ``MsgArray3D``              Directly access messages via a 3 dimensional array
```

Comment:
`MsgBucket` has been added since I first wrote this.

Reply:
https://github.com/FLAMEGPU/FLAMEGPU2/blob/master/include/flamegpu/runtime/messaging/Bucket.h